File: extract.js
const https = require('https');
const fs = require('fs');

const url = "https://www.sarkariexam.com/";

https.get(url, { headers: { 'User-Agent': 'Mozilla/5.0' } }, (res) => {
    let data = '';
    res.on('data', chunk => data += chunk);
    res.on('end', () => {
        const extracted = {};
        
        // Find breaking news
        const breakingMatch = [...data.matchAll(/<div class="brack-text-rigt"><a[^>]*href="([^"]+)"[^>]*>(.*?)<\/a><\/div>/g)];
        if (breakingMatch.length > 0) {
            extracted["Breaking News"] = breakingMatch.map(m => ({ title: m[2].replace(/<[^>]+>/g, '').trim(), url: m[1] }));
        }

        const sections = [...data.matchAll(/<h4[^>]*>.*?<strong>(.*?)<\/strong>.*?<\/h4>(.*?)<\/ul>/gis)];
        for (const match of sections) {
            const header = match[1];
            const content = match[2];
            
            const links = [];
            const linkMatches = [...content.matchAll(/<li[^>]*><a[^>]*href="([^"]+)"[^>]*>(.*?)<\/a><\/li>/gi)];
            for (const lmatch of linkMatches) {
                const title = lmatch[2].replace(/<[^>]+>/g, '').trim();
                const href = lmatch[1];
                if (title && href) {
                    links.push({ title, url: href });
                }
            }
            if (links.length > 0) {
                if (!extracted[header]) extracted[header] = [];
                extracted[header].push(...links);
            }
        }
        
        fs.writeFileSync('c:\\Users\\user\\.gemini\\antigravity\\scratch\\ranjeet-cyber-cafe\\extracted_data.json', JSON.stringify(extracted, null, 2));
        console.log('SUCCESS');
    });
}).on('error', err => console.log('ERROR:', err));

File: extract.py
import urllib.request
import re
import json

url = "https://www.sarkariexam.com/"
req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64)'})
try:
    html = urllib.request.urlopen(req).read().decode('utf-8', errors='ignore')
    
    # regex to find headers
    # <h4 class="wp-block-heading has-text-align-center"><strong>Result</strong></h4>
    pattern = r'<h4[^>]*>.*?<strong>(.*?)</strong>.*?</h4>(.*?)(?=<h4|<div class="wp-block-columns|$)'
    matches = re.findall(pattern, html, re.DOTALL | re.IGNORECASE)
    
    extracted = {}
    for header, content in matches:
        # find all list items
        links = re.findall(r'<a[^>]*href="([^"]+)"[^>]*>(.*?)</a>', content)
        if links:
            if header not in extracted:
                extracted[header] = []
            for href, title in links:
                title = re.sub(r'<[^>]+>', '', title).strip()
                extracted[header].append({"title": title, "url": href})
                
    # Also look for breaking news
    breaking = re.findall(r'<div class="brack-text-rigt"><a[^>]*href="([^"]+)"[^>]*>(.*?)</a></div>', html)
    if breaking:
        extracted["Breaking News"] = [{"title": re.sub(r'<[^>]+>', '', t).strip(), "url": h} for h, t in breaking]

    with open('c:\\Users\\user\\.gemini\\antigravity\\scratch\\ranjeet-cyber-cafe\\extracted_data.json', 'w', encoding='utf-8') as f:
        json.dump(extracted, f, indent=4)
        
    print("SUCCESS")
except Exception as e:
    print(f"ERROR: {e}")
